cloudinit/distros/package_management/snap.py

```diff
@@ -35,4 +35,23 @@ def install_packages(self, pkglist: Iterable) -> UninstalledPackages:
 
     @staticmethod
     def upgrade_packages():
-        subp.subp(["snap", "refresh"])
+        command = ["snap", "get", "system", "-d"]
+        snap_hold = None
+        try:
+            result = subp.subp(command)
+            snap_hold = (
+                util.load_json(result.stdout).get("refresh", {}).get("hold")
```

Comment:
If I start a new lxd container, I see a refresh hold immediately upon starting the container. I think a refresh hold exists regardless if the user has specified one and just points to the next time snap will refresh.

Reply:
Hrm, yes you are correct. Just saw this on jammy and oracular images.  Ok, we can't approach this feature from this angle. If someone wants temporary snap hold during boot, They can  add a runcmd: snap refresh --unhold to user-data. I'll update docs and avoid the time-based hold since it looks to be set during boot anyway that ties directly to the `Jul 10 22:33:58.917075 test-j systemd[1]: Started Snap Daemon.` getting started in journalctl.